torch/_inductor/codegen/wrapper.py

```diff
@@ -613,16 +613,15 @@ def write_prefix(self):
                 output_types = "void"
 
             if inputs_len != 0:
-                inputs_args = ["at::Tensor&"] * len(V.graph.graph_inputs.keys())
-                inputs_args = ", ".join(inputs_args)
-                inputs_args = f"std::tuple<{inputs_args}>"
+                inputs_types = "std::vector<at::Tensor>"
 
                 self.wrapper_call.writeline(
-                    f"{output_types} call_{self._call_func_id}({inputs_args} args) {{"
+                    f"{output_types} call_{self._call_func_id}({inputs_types} args) {{"
                 )
                 inputs_keys_str = ", ".join(V.graph.graph_inputs.keys())
                 self.wrapper_call.writeline(f"at::Tensor {inputs_keys_str};")
-                self.wrapper_call.writeline(f"std::tie({inputs_keys_str}) = args;")
+                for idx, input_key in enumerate(V.graph.graph_inputs.keys()):
+                    self.wrapper_call.writeline(f"{input_key} = args[{idx}];")
             else:
                 self.wrapper_call.writeline(
                     f"{output_types} call_{self._call_func_id}(std::tuple<> args) {{"
```

Comment:
How about we always do `std::vector` now? For no arg case, it could be an empty vector.

Reply:
I've simplified the code to use `std::vector` for both `inputs_len >= 1` and `inputs_len == 0`.